### DISCUSSION TITLE
Update/Refresh Chart View

### DISCUSSION THREAD

**Discussion by langejay** [Live-Charts/LiveCharts2#407] [Q&A] [answered]
I have the Charts AutoUpdateEnabled set to false to allow time for the Backend data to update.  I would like to then 'manually' request a refresh/update of the WPF view.  

The only way I could get this to work is to do a 'hack':

`        var store = Chart.AnimationsSpeed;
        Chart.AnimationsSpeed = new TimeSpan(0);
        Chart.AnimationsSpeed = store;`

The following has not worked:

Chart.UpdateLayout()
Chart.AutoUpdateEnabled = true (this doesn't trigger a refresh, I wonder if it should when set to true?)
Chart.Visibility = Show

Is there a method/function that I can use to prompt a redraw?  If not, is there a better 'hack' than the one I have used

TIA

**Comment by beto-rodriguez** [comment, marked as answer]
Try to force an update using the core method:

`UIChart.CoreChart.Update(new LiveChartsCore.Kernel.ChartUpdateParams { IsAutomaticUpdate = false, Throttling = false });`

Where `UIChart` is `Chart` in your case, I mean it should work with any control provided by the library no matter the platform you are working on.

**Reply by langejay** [reply]
Thank you!  That worked a treat.   Thanks for getting in touch.

**Comment by langejay** [comment]
Is there a way to update the Legend as well?  My use case allows users to change the colour of the stroke and fill of a data points geometry.  The point sin the chart update however the legend doesn't and retains the old visual.
